api/src/iot/tests/test_parser_cctv_camera_verkeersmanagement.py

```diff
@@ -0,0 +1,391 @@
+        },
+        'privacy_declaration': "https://www.amsterdam.nl/privacy/\
+specifieke/privacyverklaring-parkeren-verkeer-bouw/verkeersmanagement",
+        'regions': [],
+        'themes': ['Mobiliteit: auto'],
+        'type': 'Overig',  # needs to be checked why not Feature
+        'reference': '5',
+    }
+
+    expected_2 = {
+        'active_until': '2050-01-01',
+        'contains_pi_data': True,
+        'datastream': '',
+        'legal_ground': 'Verkeersmanagment in de rol van wegbeheerder.',
+        'location': {'latitude': 4.895862, 'longitude': 52.381543},
+        'location_description': None,
+        'observation_goal': 'Waarnemen van het verkeer.',
+        'owner': {
+            'name': 'camera verkeersmanagement',
+            'email': 'verkeersmanagement@amsterdam.nl',
+            'organisation': 'Gemeente Amsterdam',
+        },
+        'privacy_declaration': "https://www.amsterdam.nl/privacy/\
+specifieke/privacyverklaring-parkeren-verkeer-bouw/verkeersmanagement",
+        'regions': [],
+        'themes': ['Mobiliteit: auto'],
+        'type': 'Overig',  # needs to be checked why not Feature
+        'reference': '6',
+    }
+
+    def test_import_sensor(self, sensor_data):
+        # Basic check for importing sensor data
+        owner = import_utils.import_person(sensor_data.owner)
+        import_utils.import_sensor(sensor_data, owner)
+        assert self.actual[0] == self.expected_2
+
+    def test_import_sensor_from_cctv_camera_verkeersmanagement(self, cctvcv_data):
+        """
+        provide a dict from the cctv camera verkeersmanagement api and call
+        the parser of the parse_cctv_camera_verkeersmanagement to get a sensor.
+        call the import_sensor and expected it to be imported.
+        """
+        parser = import_utils_apis.parse_cctv_camera_verkeersmanagement
+        sensor_list = list(parser(cctvcv_data))
+        sensor = sensor_list[0]
+        person = sensor.owner
+        imported_person = import_utils.import_person(person_data=person)
+        result = import_utils.import_sensor(sensor, imported_person)
+
+        assert type(result[0]) == models.Device2  # expet a device2 object to be returned
+        assert self.actual[0] == self.expected_1
+
+
+@pytest.mark.django_db
+class TestMigrateApiData:
+    """tests for the convert_api_data function."""
+
+    @property
+    def actual(self):
+        return [
+            Device2Serializer(device).data
+            for device in models.Device2.objects.all()
+        ]
+
+    def test_convert_api_data_cctvcv_only_insert_2(self, migrated_data):
+        """
+        provide a dict from the cctv sensor crowd management api and the
+        the name of the parser which is the api_name. Expect to have two
+        sensors created and a tuple to be returned.
+        """
+
+        result = import_utils_apis.convert_api_data(
+            api_name='cctv_camera_verkeersmanagement',
+            api_data=migrated_data
```

Comment:
Klopt het dat je de al converteerde data gebruikt als argument hier?

Reply:
nee, de naam is verwarrend. Ik heb het gewijzigd naar api_data_2. dit is de tweede api_data fixture. omdat ik twee fixtures gebruik met api data, noemde ik de tweede migrated_data. deze data was bedoeld om te check of bestande data wordt geüpdate of deleted. Maar ik zie dat ik het ook voor normaal insert heb gebruikt.
Ik het nu aangepast naar api_data_2 bij all tests.